homeassistant/components/juicenet.py

```diff
@@ -0,0 +1,89 @@
+"""
+Support for Juicenet cloud.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/juicenet
+"""
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.helpers import discovery
+from homeassistant.const import CONF_ACCESS_TOKEN
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['python-juicenet==0.0.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'juicenet'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_ACCESS_TOKEN): cv.string
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up the Juicenet component."""
+    import pyjuicenet
+
+    hass.data[DOMAIN] = {}
+    hass.data[DOMAIN]['unique_ids'] = []
+    hass.data[DOMAIN]['entities'] = {}
+
+    access_token = config[DOMAIN].get(CONF_ACCESS_TOKEN)
+
+    hass.data[DOMAIN]['api'] = pyjuicenet.Api(access_token)
+
+    hass.data[DOMAIN]['entities']['sensor'] = []
+    discovery.load_platform(hass, 'sensor', DOMAIN, {}, config)
+
+    return True
+
+
+class JuicenetDevice(Entity):
+    """Represent a base Juicenet device."""
+
+    def __init__(self, device, sensor_type, hass):
+        """Initialise the sensor."""
+        self.hass = hass
+        self.device = device
+        self.type = sensor_type
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self.device.name()
+
+    def update(self):
+        """Update state of the device."""
+        self.device.update_state()
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        attributes = {}
+        if self.type == 'status':
```

Comment:
This shouldn't be in a generic JuiceNet entity. This is part of an implementation.